packages/compass-indexes/src/components/indexes-toolbar/indexes-toolbar.tsx

```diff
@@ -107,9 +106,15 @@ export const IndexesToolbar: React.FunctionComponent<IndexesToolbarProps> = ({
   readOnly, // preferences readOnly.
 }) => {
   const isSearchManagementActive = usePreference('enableAtlasSearchIndexes');
+  const mongoDBMajorVersion = parseFloat(
+    serverVersion.split('.').slice(0, 2).join('.')
+  );
```

Comment:
We use semver package for handling mongodb version comparison, you can't really rely fully on the version here being fully parseable as a number (because it's not). There are some examples in the code here that already use semver for similar checks (see `serverSupportsSearchIndexManagement` for example), please update your code to the similar pattern